lib/graphql/stitching/shaper.rb

```diff
@@ -13,14 +13,18 @@ def initialize(supergraph:, document:, raw:)
 
       def perform!
         if @result.key?("data") && ! @result["data"].empty?
-          munge_entry(@result["data"], @document.operation.selections, @supergraph.schema.query)
+          begin
+            munge_entry(@result["data"], @document.operation.selections, @supergraph.schema.query)
+          rescue InvalidNullError => e
+            @errors << { "message" => e.message}
```

Comment:
@gmac what is the pattern for errors?

Reply:
The need to be pathed, which involves passing a path argument through the tree much the way planner does. As for the errors generated, I'm actually not sure how frequently the errors are issued (source versus bubble steps). I'll peek at the Bramble bubbler tomorrow.

Reply:
As for returned errors, that can go in as a separate issue if it's ever reported as an actual problem. Tinkering with a quick schema, bubbling itself is a constraint of the graph structure and does not constitute a messaged error....

```ruby
class NullBubble < GraphQL::Schema
  class Thing1 < GraphQL::Schema::Object
    field :name, String, null: false
  end

  class Thing2 < GraphQL::Schema::Object
    field :thing1, Thing1, null: false
  end

  class Query < GraphQL::Schema::Object
    field :thing2, Thing2, null: false

    def thing2
      { thing1: { name: nil } }
    end
  end

  query Query
end

result = NullBubble.execute(query: "{ thing2 { thing1 { name } } }").to_h
# {"data"=>nil, "errors"=>[{"message"=>"Cannot return null for non-nullable field Thing1.name"}]}
```

The only reported error happens at the underlying resolver level, where a null value was first plugged into a non-null slot. We'll actually be getting these errors from the underlying locations. The only corner-case is when stitching has modified nullability of a field in the combined schema... but then, it should have only ever _loosened_ the nullability constraint, never tightened it. So, I think we're safe to say null errors will flow through from the underlying service layer.